spec/models/service_template_transformation_plan_request_spec.rb

```diff
@@ -28,6 +54,25 @@
     end
   end
 
+  describe '#validate_conversion_hosts' do
+    let(:plan) { ServiceTemplateTransformationPlan.create_catalog_item(catalog_item_options) }
+    let(:request) { FactoryGirl.create(:service_template_transformation_plan_request, :source => plan) }
+
+    context 'no conversion host exists in EMS' do
+      let(:host) { FactoryGirl.create(:host, :ext_management_system => FactoryGirl.create(:ext_management_system, :zone => FactoryGirl.create(:zone))) }
+      let(:conversion_host) { FactoryGirl.create(:conversion_host, :resource => host) }
+
+      it { expect(request.validate_conversion_hosts).to eq(false) }
+    end
+
+    context 'conversion host exists in EMS' do
+      let(:host) { FactoryGirl.create(:host, :ext_management_system => dst_ems) }
+      let(:conversion_host) { FactoryGirl.create(:conversion_host, :resource => host) }
+
+      it { expect(request.validate_conversion_hosts).to eq(false) }
+    end
+  end
```

Comment:
Can you add a test for using `vm` as conversion host?

Reply:
Done.